blocks/product-teaser/product-teaser.js

```diff
@@ -0,0 +1,151 @@
+import { readBlockConfig } from '../../scripts/aem.js';
+import { performCatalogServiceQuery, renderPrice } from '../../scripts/commerce.js';
+
+const productTeaserQuery = `query productTeaser($sku: String!) {
+  products(skus: [$sku]) {
+    sku
+    urlKey
+    name
+    addToCartAllowed
+    __typename
+    images(roles: ["small_image"]) {
+      label
+      url
+    }
+    ... on SimpleProductView {
+      price {
+        ...priceFields
+      }
+    }
+    ... on ComplexProductView {
+      priceRange {
+        minimum {
+          ...priceFields
+        }
+        maximum {
+          ...priceFields
+        }
+      }
+    }
+  }
+}
+fragment priceFields on ProductViewPrice {
+  regular {
+    amount {
+      currency
+      value
+    }
+  }
+  final {
+    amount {
+      currency
+      value
+    }
+  }
+}`;
+
+function renderPlaceholder(config, block) {
+  block.textContent = '';
+  block.appendChild(document.createRange().createContextualFragment(`
+    <div class="image">
+      <div class="placeholder"></div>
+    </div>
+    <div class="details">
+      <h1></h1>
+      <div class="price"></div>
+      <div class="actions">
+        ${config['details-button'] ? '<a href="#" class="button primary disabled">Details</a>' : ''}
+        ${config['cart-button'] ? '<button class="secondary" disabled>Add to Cart</button>' : ''}
+      </div>
+    </div>
+  `));
+}
+
+function renderImage(image, size = 250) {
+  const { url: imageUrl, label } = image;
+  const createUrlForWidth = (url, w, useWebply = true) => {
+    const newUrl = new URL(url, window.location);
+    if (useWebply) {
+      newUrl.searchParams.set('format', 'webply');
+      newUrl.searchParams.set('optimize', 'medium');
+    } else {
+      newUrl.searchParams.delete('format');
+    }
+    newUrl.searchParams.set('width', w);
+    newUrl.searchParams.delete('quality');
+    newUrl.searchParams.delete('dpr');
+    newUrl.searchParams.delete('bg-color');
+    return newUrl.toString();
+  };
+
+  const createUrlForDpi = (url, w, useWebply = true) => `${createUrlForWidth(url, w, useWebply)} 1x, ${createUrlForWidth(url, w * 2, useWebply)} 2x, ${createUrlForWidth(url, w * 3, useWebply)} 3x`;
```

Comment:
We could consider merging this with `createOptimizedPicture` from `aem.js` in the future, but let's leave as is for now.